### ISSUE TITLE
Searching docs for "typecheck" yields nothing

### ISSUE THREAD

**Issue by racket-bug-submit** [samth/test-bugs#81]
###### Originally submitted on: Sat Dec 20 17:56:01 -0500 2008

I suggest that "typecheck" be added as a search term for the
Typed Scheme documentation, esp. since the error messages from
Typed Scheme begin "typecheck:"
##### Steps to Reproduce:

type "typecheck", hit F1, no search results found.
##### Release:

4.1.3.2-svn3dec2008
##### Environment:

macosx "Darwin uccello.local 9.5.0 Darwin Kernel Version 9.5.0: Wed Sep  3 11:29:43 PDT 2008; root:xnu-1228.7.58~1/RELEASE_I386 i386 i386" (i386-macosx/3m) (get-display-depth) = 32
Human Language: english
(current-memory-use) 130842132

Collections:
(("/Users/clements/clements/collects" ".svn" "badger" "clements-lib" "contour-letters" "csc430-p6" "info-domain" "iTunes-helper" "java-helpers" "jbc-gnucash" "matrix-arithmetic" "mboxrd-read" "mnemisis" "profj-auto-runner" "projector" "shelly" "single-stepper" "smootxes" "template-tester" "txomp" "web-utilities") ("/Users/clements/matthias/collects" non-existent-path) ("/Users/clements/Library/PLT Scheme/4.1.3.2/collects" non-existent-path) ("/Users/clements/plt/collects" ".svn" "afm" "algol60" "browser" "combinator-parser" "compiler" "config" "defaults" "drscheme" "dynext" "embedded-gui" "eopl" "errortrace" "ffi" "file" "framework" "frtime" "games" "graphics" "gui-debugger" "guibuilder" "handin-client" "handin-server" "help" "hierlist" "honu" "htdch" "htdp" "html" "icons" "info-domain" "lang" "launcher" "lazy" "macro-debugger" "make" "mred" "mrlib" "mysterx" "mzcom" "mzlib" "mzscheme" "net" "openssl" "parser-tools" "planet" "plot" "preprocessor" "profj" "profjWizard" "r5rs"!
  "r6rs" "readline" "redex" "repos-time-stamp" "rnrs" "s-exp" "scheme" "scribble" "scribblings" "setup" "sgl" "sirmail" "slatex" "slideshow" "srfi" "srpersist" "stepper" "string-constants" "swindle" "syntax" "syntax-color" "teachpack" "test-box-recovery" "test-engine" "tests" "tex2page" "texpict" "trace" "typed" "typed-scheme" "version" "waterworld" "web-server" "wxme" "xml"))
Computer Language: (("Module") (#(#t write mixed-fraction-e #f #t debug) (default) #() "#lang scheme\n" #t))
###### This bug was converted from Gnats bug [9984](http://bugs.racket-lang.org/query/?cmd=view&pr=9984).

[anon-submit clements AT brinckerhoff.org]

**Comment by racket-bug-submit** [comment]
On Sat, 20 Dec 2008 22:23:56 -0500, samth@plt-scheme.org wrote:

How do I add a 'search term' to my documentation?

Thanks,
sam th

On Sat, Dec 20, 2008 at 5:56 PM,  clements@brinckerhoff.org wrote:

> A new problem report is waiting at
>  http://bugs.plt-scheme.org/query/?cmd=view&pr=9984
> 
> Reported by John Clements for release: 4.1.3.2-svn3dec2008
> 
> **\* Description:
> I suggest that "typecheck" be added as a search term for the
> Typed Scheme documentation, esp. since the error messages from
> Typed Scheme begin "typecheck:"
> 
> **\* How to repeat:
> type "typecheck", hit F1, no search results found.
> 
> **\* Environment:
> macosx "Darwin uccello.local 9.5.0 Darwin Kernel Version 9.5.0: Wed Sep  3 11:29:43 PDT 2008; root:xnu-1228.7.58~1/RELEASE_I386 i386 i386" (i386-macosx/3m) (get-display-depth) = 32
> Human Language: english
> (current-memory-use) 130842132
> 
> Collections:
> (("/Users/clements/clements/collects" ".svn" "badger" "clements-lib" "contour-letters" "csc430-p6" "info-domain" "iTunes-helper" "java-helpers" "jbc-gnucash" "matrix-arithmetic" "mboxrd-read" "mnemisis" "profj-auto-runner" "projector" "shelly" "single-stepper" "smootxes" "template-tester" "txomp" "web-utilities") ("/Users/clements/matthias/collects" non-existent-path) ("/Users/clements/Library/PLT Scheme/4.1.3.2/collects" non-existent-path) ("/Users/clements/plt/collects" ".svn" "afm" "algol60" "browser" "combinator-parser" "compiler" "config" "defaults" "drscheme" "dynext" "embedded-gui" "eopl" "errortrace" "ffi" "file" "framework" "frtime" "games" "graphics" "gui-debugger" "guibuilder" "handin-client" "handin-server" "help" "hierlist" "honu" "htdch" "htdp" "html" "icons" "info-domain" "lang" "launcher" "lazy" "macro-debugger" "make" "mred" "mrlib" "mysterx" "mzcom" "mzlib" "mzscheme" "net" "openssl" "parser-tools" "planet" "plot" "preprocessor" "profj" "profjWizard" "r5rs"!
>  "r6rs" "readline" "redex" "repos-time-stamp" "rnrs" "s-exp" "scheme" "scribble" "scribblings" "setup" "sgl" "sirmail" "slatex" "slideshow" "srfi" "srpersist" "stepper" "string-constants" "swindle" "syntax" "syntax-color" "teachpack" "test-box-recovery" "test-engine" "tests" "tex2page" "texpict" "trace" "typed" "typed-scheme" "version" "waterworld" "web-server" "wxme" "xml"))
> Computer Language: (("Module") (#(#t write mixed-fraction-e #f #t debug) (default) #() "#lang scheme\n" #t))

## 

sam th
samth@ccs.neu.edu

**Comment by racket-bug-submit** [comment]
On Sat, 20 Dec 2008 20:48:02 -0800, clements@brinckerhoff.org wrote:

On Dec 20, 2008, at 7:23 PM, Sam TH wrote:

> How do I add a 'search term' to my documentation?

Okay, I should have dug these docs up myself.

It looks like this change:

@title[#:tag "top"]{@bold{Typed Scheme}: Scheme with Static Types}

@author["Sam Tobin-Hochstadt"]

@index["typecheck"]

Typed Scheme is a Scheme-like language, with a type system that
supports common Scheme programming idioms.  Explicit type declarations
are required --- that is, there is no type inference.  The language
...

might do what I'm asking for, and I'll test it... right after I get  
finished running a build on DrScheme.

John

**Comment by racket-bug-submit** [comment]
On Sun, 21 Dec 2008 02:02:12 -0800, clements@brinckerhoff.org wrote:

On Dec 20, 2008, at 8:48 PM, John Clements wrote:

> might do what I'm asking for, and I'll test it... right after I get  
> finished running a build on DrScheme.

Yep, it looks like it worked.  I committed it, but I'll back it out if  
you don't like it.

John

**Comment by racket-bug-submit** [comment]
On Sun, 21 Dec 2008 07:01:10 -0700, mflatt@cs.utah.edu wrote:

At Sat, 20 Dec 2008 20:48:02 -0800, John Clements wrote:

> It looks like this change:
> 
> @title[#:tag "top"]{@bold{Typed Scheme}: Scheme with Static Types}
> 
> @author["Sam Tobin-Hochstadt"]
> 
> @index["typecheck"]
> 
> Typed Scheme is a Scheme-like language, with a type system that
> supports common Scheme programming idioms.  Explicit type declarations
> are required --- that is, there is no type inference.  The language
> ...
> 
> might do what I'm asking for, and I'll test it... right after I get  
> finished running a build on DrScheme.

Beware that `index' creates an element (which is roughly a``word'' in
a paragraph), so the above makes an empty paragraph. My browser ignores
an empty paragraph, but it's not really what you intended.

The `index' form is intended for pointing an index entry to a specific
word or phrase in the rendered document, as in

  The @index["self"]{@scheme[this]} identifier...

When you want an index term to go to a particular section, use
`section-index', instead. That's the right choice in the "typecheck"
case, because you want a search on "typecheck" to point to the document
as a whole.

When you use `section-index', clicking on the search result for
"typecheck" in a typical browser takes you to the TS Guide without
scrolling, while`index' causes the browser to scroll the page where
the title isn't visible. You want to see the title, and that's another
way of saying that you want `section-index' rather than`index'.

I've changed "ts-guide.scrbl", and I've also added a clarification on
`index' versus`section-index' to the Scribble docs.
